### PR TITLE
Typescript namespace fixes

### PR DESCRIPTION
**Pull request by mattheworiordan**
@funkyboy I had a quick shot at fixing the type issues.

Seems like a very simple fix as far as I can tell.  WDYT?

Should supersede https://github.com/ably/ably-js/pull/475 and fix https://github.com/ably/ably-js/issues/461

### CODE REVIEW

**Comment by mattheworiordan** [comment]
@funkyboy?  This shouldn't take more than a minute to review.

**Comment by mattheworiordan** [comment]
@jdmnd as the TS expert, are you happy with this?

File: README.md [hunk 1]
```diff
@@ -58,12 +58,18 @@ For React Native, do not use this package. Instead use the [ably-react-native](h
...
+let client = new Ably.Realtime(options); /* inferred type Ably.Realtime */
+let channel = client.channels.get('feed'); /* inferred type Ably.Types.RealtimeChannel */
 ```
 
-Additionally, the type definitions are registered with [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/ably).
+Intellisense in IDEs with TypeScript support is supported:
+
+![TypeScript suggestions](./resources/typescript-demo.gif)
+
+If your need to explicitly import the type definitions, see [ably.d.ts](./ably.d.ts).
```
Comment (jdmnd) [inline README.md]: Should say 'if you need'.

**Comment by jdmnd** [comment]
Looks good to me. It's a little unusual for the types to all live in an embedded namespace, but it does make the autocomplete a bit more intuitive.

**Comment by paddybyers** [comment]
> It's a little unusual for the types to all live in an embedded namespace, but it does make the autocomplete a bit more intuitive.

Any alternative suggestion?

File: ably.d.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
 // Definitions by: Ably <https://github.com/ably/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare namespace ablyLib {
+declare namespace Types {
   namespace ChannelState {
     type INITIALIZED = 'initialized';
     type ATTACHING = 'attaching';
```
Comment (jdmnd) [inline ably.d.ts]: This file uses double-quotes for all other strings.

**Comment by jdmnd** [comment]
> Any alternative suggestion?

Ordinarily the types would be in the top-level namespace. i.e.: 
```
import { Realtime, ClientOptions } from 'ably'
let options: ably.ClientOptions = { key: 'foo' }
```
My guess is the way it's being done currently is so that when you type `ably.` the suggestions list is short – `Realtime` is likely what you want, not `HttpPaginatedResponse`.

I think I prefer the approach used.

**Pull request merged**